spark/src/main/scala/org/apache/spark/delta/sharing/PreSignedUrlCache.scala

```diff
@@ -81,9 +92,14 @@ class CachedTableManager(
         logInfo(s"Updating pre signed urls for $tablePath (expiration time: " +
           s"${new java.util.Date(cachedTable.expiration)})")
         try {
+          val (idToUrl, expiration) = cachedTable.refresher()
           val newTable = new CachedTable(
-            preSignedUrlExpirationMs + System.currentTimeMillis(),
-            cachedTable.refresher(),
+            if (isValidUrlExpirationTime(expiration)) {
+              expiration.min(preSignedUrlExpirationMs + System.currentTimeMillis())
```

Comment:
The config `preSignedUrlExpirationMs` is because we don't know the expiration time. If the server tells us the expiration time, why do we still need to use `preSignedUrlExpirationMs`?

Reply:
makes sense, I'll just use expiration then.